sources/packages/api/src/route-controllers/restriction/restriction.controller.ts

```diff
@@ -35,10 +67,192 @@ export class RestrictionController extends BaseController {
+    @Param("studentRestrictionId") studentRestrictionId: number,
+  ): Promise<StudentRestrictionDetail> {
+    const studentRestriction =
+      await this.studentRestrictionService.getStudentRestrictionDetailsById(
+        studentId,
+        studentRestrictionId,
+      );
+    if (!studentRestriction) {
+      throw new NotFoundException("The student restriction does not exist.");
+    }
+    return {
+      restrictionId: studentRestriction.id,
+      restrictionType: studentRestriction.restriction.restrictionType,
+      restrictionCategory: studentRestriction.restriction.restrictionCategory,
+      description: studentRestriction.restriction.description,
+      createdAt: studentRestriction.createdAt,
+      updatedAt: studentRestriction.updatedAt,
+      createdBy: studentRestriction.creator
+        ? `${studentRestriction.creator.lastName}, ${studentRestriction.creator.firstName}`
+        : "",
+      updatedBy: studentRestriction.modifier
+        ? `${studentRestriction.modifier.lastName}, ${studentRestriction.modifier.firstName}`
+        : "",
+      isActive: studentRestriction.isActive,
+      restrictionNote: studentRestriction.restrictionNote?.description,
+      resolutionNote: studentRestriction.resolutionNote?.description,
+    };
+  }
+
+  /**
+   * Rest API to add a new provincial restriction to Student.
+   * Note: Federal restrictions are added/resolved by nightly job not through this API.
+   * @param userToken
+   * @param studentId
+   * @param restrictionId
+   * @param payload
+   */
+  @Groups(UserGroups.AESTUser)
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Post("/student/:studentId")
+  async addStudentProvincialRestriction(
+    @UserToken() userToken: IUserToken,
+    @Param("studentId") studentId: number,
+    @Body() payload: AddStudentRestrictionDTO,
+  ): Promise<void> {
+    const restriction =
+      await this.restrictionService.getProvincialRestrictionById(
+        payload.restriction,
+      );
+    if (!restriction) {
+      throw new UnprocessableEntityException(
+        "The given restriction type is not Provincial. Only provincial restrictions can be added.",
+      );
+    }
+    const studentRestriction = new StudentRestriction();
+    studentRestriction.student = { id: studentId } as Student;
+    studentRestriction.restriction = restriction;
+    studentRestriction.creator = { id: userToken.userId } as User;
+    if (payload.noteDescription) {
+      studentRestriction.restrictionNote = {
+        description: payload.noteDescription,
+        noteType: NoteType.Restriction,
+        creator: {
+          id: studentRestriction.creator.id,
+        } as User,
+      } as Note;
+    }
+    const updatedRestriction =
+      await this.studentRestrictionService.addProvincialRestriction(
+        studentRestriction,
+      );
+    /**mapping the note added for restriction to student notes**/
+    if (updatedRestriction.restrictionNote) {
+      await this.studentService.saveStudentNote(
```

Comment:
seems like, this transaction is dependent on line 192, maybe you create it as a single transaction

Reply:
As discussed not using transactions because if 2nd transaction fails it should not affect the first one.